### PR TITLE
open-stage-control: use postPatch for copying in package-lock.json

### PR DESCRIPTION
**Pull request by lilyinstarlight**
###### Description of changes

Follow-up to https://github.com/NixOS/nixpkgs/pull/200554#discussion_r1027035537 and #201731

Instead of generating a patch file to provide package-lock.json, this copies the file in `postPatch` because that works now since #201731 was merged

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/open-stage-control/default.nix [hunk 1]
```diff
@@ -11,10 +11,10 @@ buildNpmPackage rec {
     hash = "sha256-XgwlRdwUSl4gIRKqk6BnMAKarVvp291zk8vmNkuRWKo=";
   };
 
-  patches = [
-    # Use generated package-lock.json since upstream does not provide one in releases
-    ./package-lock.json.patch
-  ];
+  # Use generated package-lock.json since upstream does not provide one in releases
+  postPatch = ''
+    cp -T ${./package-lock.json} ./package-lock.json
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/open-stage-control/default.nix:16]: Is now ``ln -s`` working?

Comment (lilyinstarlight) [reply pkgs/applications/audio/open-stage-control/default.nix:16]: It should be!

I didn't want there to be a chance that the package-lock.json link ends up in some output and ends up having nixpkgs in the closure though

I'll do some testing and see if that actually happens or not. If not, I don't mind changing it

Comment (lilyinstarlight) [reply pkgs/applications/audio/open-stage-control/default.nix:16]: Oh yeah Nix copies the individual file for these types of references anyway so it doesn't matter

Aside from that, the testing showed that `package-lock.json` isn't packed in `npm pack` and so it isn't present in the output anyway with the default install hook or the custom phase here

So yeah I'll push the change in a moment!

**Review by figsoda — APPROVED** [review]

**Comment by dotlambda** [comment]
Can we ask upstream to add a `package-lock.json` to version control?

**Comment by lilyinstarlight** [comment]
> Can we ask upstream to add a `package-lock.json` to version control?

I've opened jean-emmanuel/open-stage-control#805

**Pull request merged**